lib/libc/stdlib/malloc/mrs/mrs.c

```diff
@@ -94,12 +94,17 @@
  *   times the size of the heap (default 4).
  * CONCURRENT_REVOCATION_PASSES: Number of concurrent revocation pass
  *   before the stop-the-world pass.
- *
- * OPTIONAL_QUARANTINING: Allow quarantining to be enabled/disabled at
- *   runtime and for mrs to fall back to not quarantining if
- *   cheri_revoke_get_shadow(2) is unimplemented (ENOSYS).
  */
 
+#define	MALLOCX_LG_ALIGN_BITS	6
+#define	MALLOCX_LG_ALIGN_MASK	((1 << MALLOCX_LG_ALIGN_BITS) - 1)
+/* Use MALLOCX_ALIGN_GET() if alignment may not be specified in flags. */
+#define	MALLOCX_ALIGN_GET_SPECIFIED(flags)				\
+    ((size_t)1 << (flags & MALLOCX_LG_ALIGN_MASK))
+/* Ignore zero and too small alignments */
+#define	MALLOCX_ALIGN_GET(flags)					\
+    (MALLOCX_ALIGN_GET_SPECIFIED(flags) & (SIZE_T_MAX - (sizeof(void *) - 1)))
```

Comment:
Technically asking for, say, an alignment of 4 is a sensible request (e.g. you want 2 bytes at a 4 byte address), so from a _decoding_ perspective I do think this should mirror jemalloc's filtering of just the LSB to strip out 2^0 == 1 as meaning no alignment specified. Then you leave it up to mrs_mallocx to decide what to do with that (which should already be correct?).